index.js

```diff
@@ -1582,22 +1630,6 @@
   function RecordType(fields) {
     var keys = Object.keys(fields).sort();
 
-    var invalidFieldNames = keys.filter(function(k) {
-      return $$type(fields[k]) !== 'sanctuary-def/Type';
-    });
-    if (!isEmpty(invalidFieldNames)) {
-      throw new TypeError(trimTrailingSpaces(
-        'Invalid values\n\n' +
-        'The argument to ‘RecordType’ must be an object ' +
-          'mapping field name to type.\n\n' +
-        'The following mappings are invalid:\n\n' +
-        Z.reduce(function(s, k) {
-          var v = fields[k];
-          return s + '  - ' + Z.toString(k) + ': ' + Z.toString(v) + '\n';
-        }, '', invalidFieldNames)
-      ));
-    }
-
```

Comment:
Replacing this with `def('RecordType', {}, [StrMap(Type), Type], RecordType)` feels very good. :)